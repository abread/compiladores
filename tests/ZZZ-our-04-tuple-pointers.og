public int og() {
	auto justneedthetype = 1, (2.0, (3.5, 4), "asd"), "def";
	auto b = justneedthetype?;
	b = [3];

	for auto i , j = 0, 1; i < 3; i = i + 1, j = j + 1 do {
		b[i]@1 = j;
		b[i]@2@1 = j + 0.1;
		b[i]@2@2@1 = j * 2.1;
		b[i]@2@2@2 = j * 2;
		b[i]@2@3 = "heh";
		b[i]@3 = "heh";
	}

	for int i = 0; i < 3; i = i + 1 do {
		write "b[", i, "]@1 = "; writeln b[i]@1;
		write "b[", i, "]@2@1 = "; writeln b[i]@2@1;
		write "b[", i, "]@2@2@1 = "; writeln b[i]@2@2@1;
		write "b[", i, "]@2@2@2 = "; writeln b[i]@2@2@2;
		write "b[", i, "]@2@3 = "; writeln b[i]@2@3;
		write "b[", i, "]@3 = "; writeln b[i]@3;
	}

	return 0;
}
