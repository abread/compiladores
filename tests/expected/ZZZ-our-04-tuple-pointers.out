b[0]@1 = 1
b[0]@2@1 = 1.1
b[0]@2@2@1 = 2.1
b[0]@2@2@2 = 2
b[0]@2@3 = heh
b[0]@3 = heh
b[1]@1 = 2
b[1]@2@1 = 2.1
b[1]@2@2@1 = 4.2
b[1]@2@2@2 = 4
b[1]@2@3 = heh
b[1]@3 = heh
b[2]@1 = 3
b[2]@2@1 = 3.1
b[2]@2@2@1 = 6.3
b[2]@2@2@2 = 6
b[2]@2@3 = heh
b[2]@3 = heh
